src/reporter/saucelabsReporter.js

```diff
@@ -0,0 +1,78 @@
+function JasmineSaucelabsReporter(config, instanceConfig, logger, collector) {
+  this.config = config;
+  this.instanceConfig = instanceConfig;
+  this.logger = logger;
+  this.collector = collector;
+}
+
+JasmineSaucelabsReporter.prototype.suiteStarted = function () {
+  browser.executeScript('sauce:context=Suite started: ' + this.collector.getCurrentSuite().name);
+};
+
+JasmineSaucelabsReporter.prototype.specStarted = function () {
+  browser.executeScript('sauce:context=Spec started: ' + this.collector.getCurrentSpec().name);
+};
+
+JasmineSaucelabsReporter.prototype.specDone = function () {
+  browser.executeScript('sauce:context=Spec done: ' + this.collector.getCurrentSpec().name);
+};
+
+JasmineSaucelabsReporter.prototype.suiteDone = function () {
+  browser.executeScript('sauce:context=Suite done: ' + this.collector.getCurrentSuite().name);
+};
+
+JasmineSaucelabsReporter.prototype.jasmineDone = function () {
+  var overview = this.collector.getOverview();
+  browser.executeScript('sauce:job-result=' + overview.status);
+};
+
+JasmineSaucelabsReporter.prototype.register = function (jasmineEnv) {
+  jasmineEnv.addReporter(this);
+
+  ['click', 'sendKeys'].forEach(this._registerOnAction.bind(this));
+  this._registerOnExpectation();
+};
+
+JasmineSaucelabsReporter.prototype._registerOnAction = function (action) {
+  var originalAction = protractorModule.parent.parent.exports.WebElement.prototype[action];
+
+  protractorModule.parent.parent.exports.WebElement.prototype[action] = function () {
+    var element = this;
+    var actionValue = arguments[0];
+
+    // TODO: save the locator which was used to find the element
```

Comment:
just an idea for the future - maybe we could define an API that the reporters can use to extract this information from the element API. Because we need it in nearly all reporters - here, in screenshots reporter, etc.